packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -355,21 +356,7 @@ export async function* OpenAIChatModel(
     : Object.entries(props.functionDefinitions).map(([functionName, functionDefinition]) => ({
         name: functionName,
         description: functionDefinition.description,
-        parameters: {
-          type: 'object',
-          required: Object.keys(functionDefinition.parameters).filter(
-            (name) => functionDefinition.parameters[name].required
-          ),
-          properties: Object.keys(functionDefinition.parameters).reduce(
-            (map: Record<string, any>, paramName) => ({
-              ...map,
-              [paramName]: {
-                type: functionDefinition.parameters[paramName].type,
-              },
-            }),
-            {}
-          ),
-        },
+        parameters: getParametersSchema(functionDefinition.parameters),
```

Comment:
This is an improvement to `FunctionCall` that both enables this PR and makes FunctionCall stronger.